src/utils/flatten.js

```diff
@@ -55,7 +56,16 @@ const flatten = (
     /* Either execute or defer the function */
     if (typeof chunk === 'function') {
       if (executionContext) {
-        ruleSet.push(...flatten([chunk(executionContext)], executionContext))
+        const nextChunk = chunk(executionContext)
+        /* Throw if a React Element was given styles */
+        if (React.isValidElement(nextChunk)) {
+          const elementName = chunk.displayName || chunk.name
+          throw new StyledError(
+            1,
+            `A plain React element (${elementName}) has been interpolated into styles, probably as a component selector (https://www.styled-components.com/docs/advanced#referring-to-other-components). Only styled-component classes can be targeted in this fashion.`
```

Comment:
Also this message should be moved to errors.md. Please see how the other ones are set up.

Reply:
Ok. Just did this. Let's go!